core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/protocol/restxml/XmlGenerator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.protocol.restxml;
+
+import java.io.StringWriter;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.util.xml.XmlWriter;
+
+@SdkInternalApi
+public interface XmlGenerator {
```

Comment:
Do we need this abstraction? The reasoning for JSON having it is to support other wire formats like ION and CBOR.